### ISSUE TITLE
1.1.262 raises Unknown Variable Type in match statements that work in 1.1.261.

### ISSUE THREAD

**Issue by dogweather** [microsoft/pyright#3736]
**Describe the bug**
This code type-checks ok in 1.1.261:

```python
def ensure_ends_with_period(text: str) -> NonemptyString:
    """
    Ensure that the string ends with a period.
    """
    match (text):
        case s if s.endswith(".") or s.endswith('."'):
            return NonemptyString(text)     # False positive error
        case s:
            return NonemptyString(s + ".")  # False positive error
```

But in 1.1.262, it produces these errors:

```
text.py:82:20 - error: Return type is unknown (reportUnknownVariableType)
text.py:84:20 - error: Return type is unknown (reportUnknownVariableType)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This was already reported [here](https://github.com/microsoft/pyright/issues/3728), and a fix is already on the way.

**Comment by dogweather** [comment]
Thanks!

**Comment by dogweather** [comment]
@erictraut The 261->262 upgrade also introduced incorrect warnings for `reportUnusedVariable` and `reportUnusedImport`. Should I create issues for either of those?

**Comment by erictraut** [comment]
Are those issues specifically related to symbols referenced within a `case` statement? If so, then it's the same issue. If not, then it could be a different issue, and a separate bug report (with repro) would be appreciated.

**Comment by dogweather** [comment]
Yep, they're all within a `case`. Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.263, which I just published.

**Comment by dogweather** [comment]
> This is addressed in pyright 1.1.263, which I just published.

Excellent, thank you.
